header-footer-grid/Core/Components/PaletteSwitch.php

```diff
@@ -0,0 +1,378 @@
+<?php
+/**
+ * Frontend Palette Switch Component class for Header Footer Grid.
+ *
+ * Name:    Header Footer Grid
+ * Author:  Bogdan Preda <bogdan.preda@themeisle.com>
+ *
+ * @version 1.0.0
+ * @package HFG
+ */
+
+namespace HFG\Core\Components;
+
+use HFG\Core\Settings\Manager as SettingsManager;
+use HFG\Main;
+use Neve\Core\Settings\Config;
+use Neve\Core\Styles\Dynamic_Selector;
+
+/**
+ * Class PaletteSwitch
+ *
+ * @package HFG\Core\Components
+ */
+class PaletteSwitch extends Abstract_Component {
+
+	const COMPONENT_ID      = 'header_palette_switch';
+	const LIGHT_PALETTE_ID  = 'light_palette';
+	const DARK_PALETTE_ID   = 'dark_palette';
+	const TOGGLE_ICON_ID    = 'toggle_icon';
+	const PLACEHOLDER_ID    = 'placeholder';
+	const AUTO_ADJUST       = 'auto_adjust_color';
+	const SIZE_ID           = 'icon_size';
+	const DEFAULT_ICON_SIZE = 16;
+
+	/**
+	 * The section icon.
+	 *
+	 * @access protected
+	 * @var string $icon
+	 */
+	protected $icon = 'buddicons-topics';
+	/**
+	 * The component default width.
+	 *
+	 * @since   1.0.0
+	 * @access  protected
+	 * @var int $width
+	 */
+	protected $width = 1;
+	/**
+	 * The component slug.
+	 *
+	 * @since   1.0.0
+	 * @access  protected
+	 * @var string $section
+	 */
+	protected $component_slug = 'hfg-palette-icon';
+
+	/**
+	 * PaletteSwitch constructor
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->set_property( 'label', __( 'Palette Switch', 'neve' ) );
+		$this->set_property( 'id', $this->get_class_const( 'COMPONENT_ID' ) );
+		$this->set_property( 'default_selector', '.builder-item--' . $this->get_id() );
+
+		add_filter( 'neve_after_css_root', [ $this, 'toggle_css' ], 10, 1 );
+		add_action( 'wp_enqueue_scripts', [ $this, 'load_scripts' ] );
+	}
+
+	/**
+	 * Load Component Scripts
+	 *
+	 * @return void
+	 */
+	public function load_scripts() {
+		wp_add_inline_script( 'neve-script', $this->toggle_script() );
+	}
+
+	/**
+	 * Get JS contents from file to use as inline script
+	 *
+	 * @return string
+	 */
+	public function toggle_script() {
+		$script = '';
+		global $wp_filesystem;
+		require_once ABSPATH . 'wp-admin/includes/file.php';
+		WP_Filesystem();
+		$file            = NEVE_MAIN_DIR . 'header-footer-grid/assets/js/components/palette_switch.js';
+		$is_file_present = $wp_filesystem->exists( $file );
+		if ( $is_file_present ) {
+			$contents = $wp_filesystem->get_contents( $file );
+			if ( $contents ) {
+				$script = $contents;
+			}
+		}
+		return $script;
+	}
+
+	/**
+	 * Methods used to filter CSS global colors
+	 *
+	 * @param string $css The CSS string.
+	 *
+	 * @return string
+	 */
+	public function toggle_css( $css ) {
+		$css .= ' ';
+
+		$auto_adjust = get_theme_mod( $this->get_id() . '_' . self::AUTO_ADJUST, 1 );
+
+		$customizer       = get_theme_mod( 'neve_global_colors', neve_get_global_colors_default( true ) );
```

Comment:
@preda-bogdan I think we should check if the palette still exists. A user might delete the palette after setting up the toggle with it in use. Maybe we should treat that case by falling back on the `darkMode` if it's not there.


@JohnPixle @selul Do we need two selectors for the palettes? Doesn't this kind of override the Global colors selected palette? I think we could do something like use the currently selected one for the light color, and only allow the user to pick the second one. Does it make sense?

Reply:
@abaicus thank you for pointing out this, I added some defaults for now in case the selected  color is not available anymore. I also think we can use only one palette selector to define the contrast palette and use the global defined as a base palette.

Reply:
> @JohnPixle @selul Do we need two selectors for the palettes? Doesn't this kind of override the Global colors selected palette? I think we could do something like use the currently selected one for the light color, and only allow the user to pick the second one. Does it make sense?

Yes, it does, this is what I pointed out in the [last comment](https://github.com/Codeinwp/neve/pull/2783#issuecomment-825453098)

I believe that the currently selected palette in the customiser should always be used in the switcher. And users simply choose one of the other palettes to be used as their dark variant. I think this is what you suggesting too @abaicus  ?

Reply:
@JohnPixle Yes, that is what I meant 👍🏻 I think it would make more sense 👯

Reply:
Let's do this too, @preda-bogdan

Reply:
cc: @selul, @abaicus @JohnPixle 
I've removed the light selector and I use the active palette as the light one.

Reply:
@JohnPixle can we update the Figma design regarding this change?

Reply:
@preda-bogdan Did a quick re-work on this, please see the latest change in [figma](https://www.figma.com/file/LBW82Rm2SA5rvvLdKrRnBl/Neve-Core-UI?node-id=792%3A20988). 

We can make use of help text only in this case, and link to the color palettes.

![Screenshot 2021-04-27 at 12 27 38](https://user-images.githubusercontent.com/4193672/116219611-7790c580-a754-11eb-8e9d-d133c4a2b5bc.png)

Reply:
@JohnPixle I've updated the control.
![image](https://user-images.githubusercontent.com/23024731/116379810-0ae0fe00-a81c-11eb-8010-add874332ee9.png)

cc: @selul